data_structures/binary_tree/transform_bst_sum_tree.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+
+class Node:
+    """
+    prints the inorder Traversal of transformed tree
+    >>> sum = 0
+    >>> root = Node(11)
+    >>> root.left = Node(2)
+    >>> root.right = Node(29)
+    >>> root.left.left = Node(1)
+    >>> root.left.right = Node(7)
+    >>> root.right.left = Node(15)
+    >>> root.right.right = Node(40)
+    >>> root.right.right.left = Node(35)
+    >>> printInorder(root)
+    1 2 7 11 15 29 35 40
+
+    >>> transformTree(root)
+
+    >>> printInorder(root)
+    139 137 130 119 104 75 40 0
+    """
+
+    def __init__(self, number: int) -> None:
+        self.data = number
+        self.left = None
+        self.right = None
+
+
```

Comment:
Please create a `BinaryTree` class that implements:
* [ ] `.__iter__()` that steps thru all nodes yielding or returning their data
* [ ] `.__str__()` that uses iter() to replace `print_inorder()` with `print(binary_tree)`
* [ ] `.build_a_tree()` that does what lines 8 thru 15 do and then returns `root`.  All tests can then use this method.